packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts

```diff
@@ -326,11 +326,15 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
         const value = payment.value()
         if (!value.to) throw new Error('missing_to_from_custodial')
         if (!value.amount) throw new Error('missing_amount_from_custodial')
+        if (!value.selection) throw new Error('missing_selection_from_custodial')
         yield call(
           api.withdrawSBFunds,
-          utils.bch.toCashAddr(value.to[0].address),
+          utils.bch.isCashAddr(value.to[0].address)
+            ? value.to[0].address
+            : utils.bch.toCashAddr(value.to[0].address),
```

Comment:
Instead of checking first for the address to be a cashaddr then running `toCashAddr` here, could you just  run `utils.bch.toCashAddr(value.to[0].address)` for any possible value of `value.to[0].address`?

Reply:
looks like you would have to add the `isCashAddr` check to the to the `toCashAddr` util which may not be what you want to do so nvm.

Reply:
I want to rewrite all this crap anyway at some point.